### PR TITLE
KeyboardAvoidingView measures frame using absolute window position

### PR DESCRIPTION
**Pull request by notjosh**
Hello, friends!

I was noticing weird measurement behaviour using `KeyboardAvoidingView`.

Our setup has our `RCTRootView` within a `UINavigationController`, which we've set up to not allow content underneath. When measuring the keyboard frame, after some trial and error, the value was 64pt off - coincidentally the height of the `UINavigationBar` + status bar.

I've wrapped the calculations in a call to `measureInWindow`, which gives the expected result.

Pictures are worth a thousand words, so here's how it was before:

screen-shot-2016-11-25-at-2 36 10-pm

And after:

screen-shot-2016-11-25-at-2 36 25-pm

You can see the blue container stretching under the keyboard by 64pt (!), even though it should be keyboard-avoided (with margin 10pt). The second image shows that behaviour working correctly.

### CODE REVIEW

File: Libraries/Components/Keyboard/KeyboardAvoidingView.js [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 const Keyboard = require('Keyboard');
 const LayoutAnimation = require('LayoutAnimation');
+const NativeMethodsMixin = require('react/lib/NativeMethodsMixin');
```
Comment (notjosh) [inline Libraries/Components/Keyboard/KeyboardAvoidingView.js]: Oh, this changed recently in 3683beb88ab3233038ac32b9c33112d537478088. Addressed in b3485a8.

**Comment by lacker** [comment]
Could you add a test that covers this behavior?

**Comment by hramos** [comment]
@notjosh any chance you can add a test?

**Comment by Kudo** [comment]
@notjosh as you put RCTRootView under UINavigationController, maybe you should try to use `keyboardVerticalOffset` property, e.g. `keyboardVerticalOffset={64}` in your case.

Please refer to http://facebook.github.io/react-native/releases/0.42/docs/keyboardavoidingview.html#keyboardverticaloffset

> keyboardVerticalOffset: number
This is the distance between the top of the user screen and the react native view, may be non-zero in some use cases.

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/keyboardavoidingview-measures-frame-using-absolute-window-position

**Comment by shergin** [comment]
@notjosh Have you tested it on Android by any chance?

**Comment by notjosh** [comment]
> Have you tested it on Android by any chance?

I haven't. I haven't tested it on iOS for a while either, so I'm not sure if this is even still relevant.

**Pull request closed without merging**